File: src/main/java/com/kaala/java/sort/BubbleSort.java
package com.kaala.java.sort;

import java.util.Arrays;
import java.util.List;

/**
 * @author kaala on 8/22/16.
 */
public class BubbleSort {
    public static List<Integer> sort(List<Integer> array) {

        int temp;
        for (int i = 1; i < array.size(); i++) {
            for (int j = 0; j < array.size() - i; j++) {
                if (array.get(j) > array.get(j + 1)) {
                    temp = array.get(j);
                    array.set(j, array.get(j + 1));
                    array.set(j + 1, temp);
                }
            }
        }

        return array;
    }

    public static void main(String[] args) {
        List<Integer> list = Arrays.asList(6, 2, 4, 1, 8);
        System.out.println(sort(list));
    }
}

File: src/main/java/com/kaala/kotlin/sort/BubbleSort.kt
package com.kaala.kotlin.sort

/**
 * @author Binod Shrestha <binodshrestha@lftechnology.com> on 5/27/17.
 */

fun bubbleSort(array: IntArray): IntArray {
    var temp: Int
    (1..array.size - 1).forEach {
        i ->
        (0..array.size - i - 1).forEach {
            j ->
            if (array.get(j) > array[j + 1]) {
                temp = array[j]
                array[j] = array.get(j + 1)
                array[j + 1] = temp
            }

        }
    }

    return array
}

fun main(args: Array<String>) {
    bubbleSort(intArrayOf(6, 2, 4, 1, 8, 0, 23, 12, 45)).forEach { println(it) }
}